src/app/lib/presentation/diagnostic/diagnostic_overlay.dart

```diff
@@ -0,0 +1,103 @@
+import 'dart:async';
+
+import 'package:app/business/diagnostics/diagnostics_service.dart';
+import 'package:app/presentation/diagnostic/diagnostic_button.dart';
+import 'package:app/presentation/diagnostic/expanded_diagnostic_page.dart';
+import 'package:flutter/material.dart';
+import 'package:get_it/get_it.dart';
+
+final class DiagnosticOverlay extends StatefulWidget {
+  const DiagnosticOverlay({super.key});
+
+  @override
+  State<StatefulWidget> createState() {
+    return _DiagnosticOverlay();
+  }
+}
+
+final class _DiagnosticOverlay extends State<DiagnosticOverlay> {
+  final _diagnosticService = GetIt.I<DiagnosticsService>();
+
+  @override
+  void initState() {
+    super.initState();
+    _init();
+  }
+
+  Future<void> _init() async {
```

Comment:
Since `init` is such a small operation it should be fine but does it create a flicker when loading in?

Reply:
I dont see how it could possibly cause a flicker. It starts collapsed and becomes visible after  _init is done. It would at most appear a bit later.

Reply:
That's why I'm wondering if it's flickering by being blank for a micro second and then showing up?

Reply:
I didnt notice anything